feat: add RE for some shader properties, utility shader, nitrishape and shader emitters

FlayaN opened the pull request:
#### Add RE for:
- BSDistantTreeShaderProperty
- BSGrassShaderProperty
- BSParticleShaderRainEmitter
- BSParticleShaderSnowEmitter
- BSSkyShaderProperty
- BSUtilityShader
- NiGeometry
- NiTriBasedGeometry
- NiTriShape

#### Update RE for:
- BSBatchRenderer
  - This was fetched from community-shaders (https://github.com/doodlum/skyrim-community-shaders/blob/dev/src/Features/Skylighting.cpp#L450)
- BSShaderProperty

#### Added funcs to:
- Precipitation
	- https://github.com/doodlum/skyrim-community-shaders/blob/b10bd3e110d9e626d1626b3feb24e92e85da451d/src/Features/Skylighting.cpp#L489C13-L489C36
- BSShaderProperty

FlayaN committed:
feat: RE NiTriShape

FlayaN committed:
feat: add RE for some shader properties, utility shader and shader emitters

FlayaN commented:
~Draft until I've checked the sizes for the new REd classes~ Verified all sizes by checking dtor size, and added unk members for some

powerof3 commented:
Class member names should be camelcase without underscores (eg. `m_GeometryGroups` -> `geometryGroups`) in line with rest of CLIB

FlayaN commented:
> Class member names should be camelcase without underscores (eg. `m_GeometryGroups` -> `geometryGroups`) in line with rest of CLIB

Nice catch! Fixed

Also fixed args not having `a_` prefix
